### ISSUE TITLE
Session fails to get restored after resume on HTC

### ISSUE THREAD

**Issue by bluemarvin** [MozillaReality/FirefoxReality#2880]
## Configuration

Firefox Reality version: v9
Firefox Reality build ID: c64365e3783a6124cbe49ce3eb8661653669a980

Hardware: HTC

## Steps to Reproduce

1. Launch FxR
2. Pause
3. Resume
4. Goto 2.

## Current Behavior

Eventually [`onTrimMemory()`](https://github.com/MozillaReality/FirefoxReality/blob/master/app/src/common/shared/org/mozilla/vrbrowser/VRBrowserActivity.java#L649) will get called. After that, the GeckoSession is not restored when FxR is resumed. This manifest as a frozen browser window. The image contained in browser window contains the surface from before the `onTrimMemory()` call. Bringing up the history or bookmark menu and then hiding it will cause the GeckoSession to be restored.

## Expected Behavior

The active GeckoSession gets restored properly and the browser window is not frozen.

**Comment by bluemarvin** [comment]
I have been able to reproduce this with v8.1 on both Focus and Focus Plus. I have not had time to try older versions. I was not able to reproduce the issue on v1.3.2 which is the version included with the HTC ROM.

**Comment by bluemarvin** [comment]
This may have been fixed by https://github.com/MozillaReality/FirefoxReality/pull/2884

**Comment by Softvision-GeluHaiduc** [comment]
I have verified this issue with the latest Firefox Reality version 9 RC6 (abe931f6) installed on the HTC Vive Focus Plus headset.
Unfortunately the issue is still reproducible, the browser windows freezes and stay frozen even after inputting a different webpage in the URL bar. [Here](https://drive.google.com/open?id=1p_yoGwmRvrD401Qnsn1IModvfvLgmfrO) you can find a video recording of the issue.

**Comment by Softvision-RemusDranca** [comment]
I have retested this issue and it's no longer reproducible with the Firefox Reality version 9 rc7 (b68f946). The browser didn't froze after multiple attempts of pausing and resuming the app.
However, I came across a new behavior while trying to reproduce this issue where the video's page is wrongly displayed in immersive mode when quitting and resuming the app while in immersive mode. This behavior is documented in issue #2906.

Tested on HTC Vive Focus Plus (Android 7.1.2) and HTC Vive Focus (Android 7.1.2).
